File: src/main/kotlin/ympa_kotlin_wiremock_server/models/GetSuggestedOfferMappingEntriesResponse.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

data class GetSuggestedOfferMappingEntriesResponse(
    @field:JsonProperty("status")
    val status: ApiResponseStatusType? = null,

    @field:JsonProperty("result")
    val result: OfferMappingSuggestionsListDTO? = null,

)

File: src/App/DTO/GetSuggestedOfferMappingEntriesResponse.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * Ответ со списком рекомендованных карточек товара.
 */
class GetSuggestedOfferMappingEntriesResponse
{
    /**
     * @DTA\Data(field="status", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\ApiResponseStatusType::class})
     * @DTA\Validator(name="TypeCompliant", options={"type":\App\DTO\ApiResponseStatusType::class})
     * @var \App\DTO\ApiResponseStatusType|null
     */
    public $status;

    /**
     * @DTA\Data(field="result", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\OfferMappingSuggestionsListDTO::class})
     * @DTA\Validator(name="TypeCompliant", options={"type":\App\DTO\OfferMappingSuggestionsListDTO::class})
     * @var \App\DTO\OfferMappingSuggestionsListDTO|null
     */
    public $result;

}
